src/Compilers/VisualBasic/Portable/Binding/InitializerSemanticModel.vb

```diff
@@ -133,7 +133,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Throw ExceptionUtilities.UnexpectedValue(Me.MemberSymbol.Kind)
             End Select
 
-            Return boundInitializer
+            Dim expressionInitializer = TryCast(boundInitializer, BoundExpression)
+            Return If(expressionInitializer IsNot Nothing AndAlso expressionInitializer.Kind <> BoundKind.BadExpression, New BoundEqualsValue(initializer, expressionInitializer), boundInitializer)
```

Comment:
Yes, testing for bad expressions here seems unsound. Without this, there is a failure in a flow analysis test that includes a lambda definition in the default value expression of a parameter. Teaching the flow analysis logic about BoundEqualsValue nodes is probably necessary if checking for bad expressions is to be removed.